lib/adBlockRustUtils.js

```diff
@@ -165,14 +165,47 @@ const sampleUnblockedNetworkRequests = [
 /*
  * Throw an error if the list is blocking any of the resources from `sampleNetworkRequests`.
  */
-const sanityCheckList = ({ title, format, data }) => {
-  const filterSet = new FilterSet()
-  filterSet.addFilters(data.split('\n'), { format })
-  const engine = new Engine(filterSet)
-  for (const request of sampleUnblockedNetworkRequests) {
-    const result = engine.check(request[0], request[1], request[2])
-    if (result) {
-      throw new Error(title + ' failed sanity check for ' + request + '. Check for corrupted list contents.')
+const sanityCheckList = async ({ title, format, data }) => {
+  const dataSplit = data.split('\n')
+  {
+    const filterSet = new FilterSet()
+    filterSet.addFilters(dataSplit, { format })
+    const engine = new Engine(filterSet)
+    for (const request of sampleUnblockedNetworkRequests) {
+      const result = engine.check(request[0], request[1], request[2])
+      if (result) {
+        throw new Error(title + ' failed sanity check for ' + request + '. Check for corrupted list contents.')
+      }
+    }
+  }
+
+  // Also, attempt to convert the rules to iOS content blocking syntax via adblock-rust and see if they are valid as per webkit-content-filter-validator.
+  {
+    const filterSet = new FilterSet(true)
+    filterSet.addFilters(dataSplit, { format })
+    const { contentBlockingRules } = filterSet.intoContentBlocking()
+    const response = await fetch('https://2sq625b43mykl2tqumoue7j4k40vkupj.lambda-url.us-west-2.on.aws', {
+      body: JSON.stringify(contentBlockingRules),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      method: 'POST',
+    })
+    if (response.status === 400) {
+      const reason = await response.text()
+      if (reason === '') {
+        // Weird lambda-specific issue? Intermittently, responses will be empty 400.
+        // Ignore it for now.
```

Comment:
I've noticed the lambda will _sometimes_ return an empty `400` response, but the same payload will always succeed when provided to a local `webkit-content-filter-validator` binary. 

I'm ignoring those failures for now, but it'd be nice to fix them if possible.